client/v3/balancer/balancer.go

```diff
@@ -191,6 +214,13 @@ func (bb *baseBalancer) HandleResolvedAddrs(addrs []resolver.Address, err error)
 			// (DO NOT) delete(bb.scToSt, sc)
 		}
 	}
+
+	return nil
```

Comment:
@ptabor I think we could do better here than a `nil`. For example, every time there is a
```go
			if err != nil {
				bb.lg.Warn("NewSubConn failed", zap.String("picker", bb.picker.String()), zap.String("balancer-id", bb.id), zap.Error(err), zap.String("address", addr.Addr))
				continue
			}
```
above we could append the `err` to a slice and then coalesce that slice here into a non-`nil` return if desired.

(An early exit after the `bb.lg.Warn()` would be a breaking change so coalescing would be the safe way to do it.)

Reply:
Agree. We should both log them and collect. 

As long as we don't wrap them into [ErrBadResolverState](
https://pkg.go.dev/google.golang.org/grpc@v1.29.1/balancer#V2Balancer) it should have not semantic impact. 

Mid-term: I would consider throwing ErrBadResolverState in case there is no SubConn resolved, but this requires separate testing. Maybe worth adding a TODO.